components/brave_ads/core/internal/ads_impl.h

```diff
@@ -147,6 +143,10 @@ class AdsImpl final : public Ads {
   // state.
   GlobalState global_state_;
 
+  // Periodically handles database maintenance tasks, such as purging and
+  // vacuuming.
+  std::unique_ptr<database::Maintenance> database_maintenance_;
```

Comment:
we can possibly have this as  `database::Maintenance database_maintenance_;`.

Reply:
Fixed. And will follow up with transitioning database maintenance to AdsCore.